Horizon/Horizon/Sources/Features/Inbox/MessageDetails/HMessageDetailsViewModel.swift

```diff
@@ -0,0 +1,304 @@
+                    subject: conversation.subject,
+                    body: self.reply,
+                    recipientIDs: recipientIDs,
+                    attachmentIDs: attachmentViewModel?.items.compactMap { $0.id } ?? [],
+                    conversationID: conversation.id,
+                    bulkMessage: true
+                )
+            ).sink(
+                receiveCompletion: { _ in
+                    performUIUpdate {
+                        self.isSending = false
+                        self.reply = ""
+                        self.composeMessageInteractor?.cancel()
+                    }
+                },
+                receiveValue: { _ in }
+            )
+            .store(in: &self.subscriptions)
+        }
+    }
+
+    // MARK: - Private Methods
+    private func listenForAnnouncements() {
+        announcementsInteractor?.messages.sink { [weak self] announcements in
+            self?.headerTitle = announcements.first(where: { $0.id == self?.announcementID })?.title ?? String(localized: "Announcement", bundle: .horizon)
+            self?.messages = announcements
+                .filter { $0.id == self?.announcementID }
+                .map { announcement in
+                    .init(announcement: announcement)
+                }
+        }
+        .store(in: &subscriptions)
+    }
+
+    private func listenForMessages() {
+        messageDetailsInteractor?
+            .messages
+            .map(markMessageAsRead)
+            .map(sortOldestToNewest)
+            .map(toViewModels)
+            .sink { [weak self] conversationMessages in
+                guard let self = self else { return }
+                self.messages = conversationMessages
+            }
+            .store(in: &subscriptions)
+    }
+
+    private func listenForSubject() {
+        announcementsInteractor?.messages.sink { [weak self] announcements in
+            let announcementTitle = announcements.first(where: { $0.id == self?.announcementID })?.title
+            self?.headerTitle = announcementTitle ?? String(localized: "Announcement", bundle: .horizon)
+        }
+        .store(in: &subscriptions)
+
+        messageDetailsInteractor?.conversation.sink { [weak self] conversations in
+            self?.headerTitle = conversations.first { $0.id == self?.conversationID }?.subject ?? String(localized: "Conversation", bundle: .horizon)
+        }
+        .store(in: &subscriptions)
+    }
+
+    private func markMessageAsRead(_ conversationMessages: [ConversationMessage]) -> [ConversationMessage] {
+        if let conversationID = conversationID, isMarkedAsRead == false {
+            messageDetailsInteractor?.updateState(
+                messageId: conversationID,
+                state: .read
+            )
+            .sink(
+                receiveCompletion: { _ in },
+                receiveValue: { _ in }
+            )
+            .store(in: &self.subscriptions)
+            isMarkedAsRead = true
```

Comment:
Shouldn't `isMarkedAsRead = true` this be part of `receiveValue` closure? Otherwise it would get executed immediately as `markMessageAsRead` is called.